components/classmap/default/module.mjs

```diff
@@ -48,41 +52,65 @@ export const createModule = ({
     estree,
     entities,
     infos,
-    paths: new Set(),
+    references: new Set(),
   };
 };
 
-const isFunctionEstree = ({ type }) =>
-  type === "ArrowFunctionExpression" ||
-  type === "FunctionExpression" ||
-  type === "FunctionDeclaration";
-
-export const lookupModuleClosure = ({ estree, infos, paths }, position) => {
-  const maybe_path = lookupEstreePath(estree, isFunctionEstree, position);
-  if (maybe_path === null) {
-    logInfo(
-      "Could not find a function in %j at %j, will treat it as excluded.",
-      estree.loc.filename,
-      position,
-    );
-    return null;
-  } else {
-    /* c8 ignore start */
-    const info = infos.has(maybe_path) ? infos.get(maybe_path) : null;
+export const lookupModuleClosure = (
+  { estree: { filename }, infos, references },
+  position,
+) => {
+  const position_string = stringifyPosition(position);
+  if (infos.has(position_string)) {
+    const info = infos.get(position_string);
     if (info !== null) {
-      paths.add(maybe_path);
+      references.add(position_string);
     }
     return info;
-    /* c8 ignore stop */
+  } else {
+    let best_distance = Infinity;
+    let best_position = null;
+    let best_info = null;
+    for (const [candidate_position_string, candidate_info] of infos) {
+      const candidate_position = parsePosition(candidate_position_string);
+      const candidate_distance = measurePositionDistance(
+        position,
+        candidate_position,
+      );
+      if (candidate_distance < best_distance) {
+        best_distance = candidate_distance;
+        best_position = candidate_position;
+        best_info = candidate_info;
+      }
```

Comment:
Should this have some kind of max cutoff, to avoid pointing at an entirely different function in case of borked map?

Reply:
Good idea! Rebased in 54bf1f58c0400844502ca63bed8638a5defcc13f. I picked 5 lines and 50 columns as cutoff. Let's see if this appropriate.